ori/volume/server/volume_list.go

```diff
@@ -18,64 +18,39 @@ import (
 	"context"
 	"fmt"
 
-	volumev1alpha1 "github.com/onmetal/cephlet/ori/volume/api/v1alpha1"
-	"github.com/onmetal/cephlet/ori/volume/apiutils"
-	"github.com/onmetal/onmetal-api/api/storage/v1alpha1"
-	"github.com/onmetal/onmetal-api/broker/common"
+	"github.com/go-logr/logr"
 	ori "github.com/onmetal/onmetal-api/ori/apis/volume/v1alpha1"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
-	corev1 "k8s.io/api/core/v1"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/labels"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func (s *Server) aggregateCephVolume(
-	pvc *corev1.PersistentVolumeClaim,
-	getVolumeClass func(string) (*v1alpha1.VolumeClass, error),
-) (*AggregateCephVolume, error) {
-	volumeClass, err := getVolumeClass(apiutils.GetVolumeClass(pvc))
-	if err != nil {
-		return nil, fmt.Errorf("error getting onmetal volume access secret: %w", err)
-	}
-
-	return &AggregateCephVolume{
-		Pvc:         pvc,
-		VolumeClass: volumeClass,
-	}, nil
-}
-
-func (s *Server) getAggregateCephVolume(ctx context.Context, id string) (*AggregateCephVolume, error) {
-	pvc := &corev1.PersistentVolumeClaim{}
-	if err := s.getManagedAndCreated(ctx, id, pvc); err != nil {
-		if !apierrors.IsNotFound(err) {
-			return nil, fmt.Errorf("error getting onmetal volume %s: %w", id, err)
-		}
-		return nil, status.Errorf(codes.NotFound, "volume %s not found", id)
-	}
+func (s *Server) getOriVolume(ctx context.Context, log logr.Logger, id string) (*ori.Volume, error) {
 
-	return s.aggregateCephVolume(pvc, s.clientGetVolumeClassFunc(ctx))
-}
+	cephImage := &Image{}
 
-func (s *Server) getVolume(ctx context.Context, id string) (*ori.Volume, error) {
-	onmetalVolume, err := s.getAggregateCephVolume(ctx, id)
+	found, err := s.provisioner.GetCephImage(ctx, id, cephImage)
 	if err != nil {
 		return nil, err
 	}
 
-	return s.convertAggregateCephVolume(ctx, onmetalVolume)
+	if !found {
+		log.V(1).Info("Inconsistent state: Failed to find image", "id", id)
+		return nil, fmt.Errorf("failed to find image")
```

Comment:
Why is this 'inconsistent'? What if an id actually does not exist? Shouldn't we report this back?